deeppavlov/core/models/lr_scheduled_tf_model.py

```diff
@@ -0,0 +1,387 @@
+        #self._lr_var = tf.placeholder(tf.float32, shape=[], name='learning_rate')
+        self._lr_var = tf.Variable(self._lr or 0., dtype=tf.float32, name='learning_rate')
+
+        if (momentum is None) and\
+                self._optimizer not in (tf.train.AdagradDAOptimizer,
+                                        tf.train.AdagradOptimizer,
+                                        tf.train.GradientDescentOptimizer,
+                                        tf.train.ProximalGradientDescentOptimizer,
+                                        tf.train.ProximalAdagradOptimizer):
+            momentum = 0.9
+        start_val, end_val = momentum, None
+        if isinstance(momentum, (tuple, list)):
+            start_val, end_val = momentum
+        dec_type, extra = momentum_decay, None
+        if isinstance(momentum_decay, (tuple, list)):
+            dec_type, extra = momentum_decay
+
+        self._mom = start_val
+        num_it, self._mom_update_on_batch = momentum_decay_epochs, False
+        if momentum_decay_batches > 0:
+            num_it, self._mom_update_on_batch = momentum_decay_batches, True
+
+        self._mom_schedule = DecayScheduler(start_val=start_val, end_val=end_val,
+                                            num_it=num_it, dec_type=dec_type,
+                                            extra=extra)
+        # self._mom_var = tf.placeholder_with_default(0.9, shape=[], name='momentum')
+        # self._mom_var = tf.placeholder(tf.float32, shape=[], name='momentum')
+        self._mom_var = tf.Variable(self._mom or 0., dtype=tf.float32, name='momentum')
+
+        self._learning_rate_drop_patience = learning_rate_drop_patience
+        self._learning_rate_drop_div = learning_rate_drop_div
+        self._learning_rate_cur_impatience = 0.
+        self._learning_rate_last_impatience = 0.
+        self._learning_rate_cur_div = 1.
+        self._clip_norm = clip_norm
+        self._fit_batch_size = fit_batch_size
+        self._fit_learning_rate = fit_learning_rate
+        self._fit_learning_rate_div = fit_learning_rate_div
+        self._fit_beta = fit_beta
+        self._fit_min_batches = fit_min_batches
+        self._fit_max_batches = fit_max_batches
+        self._external_lr = False
+        self._external_mom = False
+
+    def load(self, exclude_scopes: Optional[Iterable] = ('Optimizer',
+                                                         'learning_rate',
+                                                         'momentum')):
+        return super().load(exclude_scopes=exclude_scopes)
+
+    def fit(self, *args):
+        data = list(zip(*args))
+        self.save()
+        if self._fit_batch_size is None:
+            raise ConfigError("in order to use fit() method"
+                              " set `fit_batch_size` parameter")
+        bs = int(self._fit_batch_size)
+        data_len = len(data)
+        num_batches = self._fit_max_batches or ((data_len - 1) // bs + 1)
+
+        avg_loss = 0.
```

Comment:
No, because there is a bias correction implementation.
See https://github.com/deepmipt/DeepPavlov/blob/dd0c79da51d81afc3a04740b7364f87740c27a07/deeppavlov/core/models/lr_scheduled_tf_model.py#L248-L249
When `avg_loss` equals `0`, after 248th line it will be equal to `(1 - self._fit_beta)*report['loss']`.
249th line removes the `(1 - self._fit_beta)` multiplier before `report['loss']`

Reply:
let the current `report['loss']` be equal to 1, then you will write `(1 - 0.98) * 1 = 0.02` to the `avg_loss`. This means that you are starting from the 0 and to reach 1 you will need a lot of steps.